meteor/client/ui/RundownView/RundownTiming.tsx

```diff
@@ -889,28 +1036,31 @@ interface ISegmentDurationProps {
 export const SegmentDuration = withTiming<ISegmentDurationProps, {}>()(
 	class SegmentDuration extends React.Component<WithTiming<ISegmentDurationProps>> {
 		render() {
-			if (
+			let duration: number | undefined = undefined
+			if (this.props.budgetDuration !== undefined) {
+				duration = this.props.budgetDuration - this.props.livePosition
+			} else if (
 				this.props.partIds &&
 				this.props.timingDurations.partExpectedDurations &&
 				this.props.timingDurations.partPlayed
 			) {
 				let partExpectedDurations = this.props.timingDurations.partExpectedDurations
 				let partPlayed = this.props.timingDurations.partPlayed
 
-				const duration = this.props.partIds.reduce((memo, partId) => {
+				duration = this.props.partIds.reduce((memo, partId) => {
 					const pId = unprotectString(partId)
 					return partExpectedDurations[pId] !== undefined
-						? memo + Math.max(0, partExpectedDurations[pId] - (partPlayed[pId] || 0))
+						? memo + partExpectedDurations[pId] - (partPlayed[pId] || 0)
```

Comment:
This change means that the `partExpectedDurations[pId]` can be negative - is that correct?

Reply:
This allows the segment duration to be negative if `partPlayed[pId]` is greater than `partExpectedDurations[pId]`